s390x: Enable exportproxy and exportserver image

jschintag opened the pull request:
Both x86_64 image versions during the 1.4.0-alpha release contained s390x binaries. Enable to build actual images for the architecture to ensure this does not happen again.

<!--  Thanks for sending a pull request!  Here are some tips for you:
1. Consider creating this PR as draft: https://github.com/kubevirt/kubevirt/blob/main/CONTRIBUTING.md#consider-opening-your-pull-request-as-draft
2. Follow the instructions for writing a release note from k8s: https://git.k8s.io/community/contributors/guide/release-notes.md
-->

### What this PR does
Before this PR:

After this PR:

<!-- (optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*: -->
Fixes #

### Why we need it and why it was done in this way
The following tradeoffs were made:

The following alternatives were considered:

Links to places where the discussion took place: <!-- optional: slack, other GH issue, mailinglist, ... -->

### Special notes for your reviewer

<!-- optional -->

### Checklist

This checklist is not enforcing, but it's a reminder of items that could be relevant to every PR.
Approvers are expected to review this list.

- [x] Design: A [design document](https://github.com/kubevirt/community/tree/main/design-proposals) was considered and is present (link) or not required
- [x] PR: The PR description is expressive enough and will help future contributors
- [x] Code: [Write code that humans can understand](https://en.wikiquote.org/wiki/Martin_Fowler#code-for-humans) and [Keep it simple](https://en.wikipedia.org/wiki/KISS_principle)
- [x] Refactor: You have [left the code cleaner than you found it (Boy Scout Rule)](https://learning.oreilly.com/library/view/97-things-every/9780596809515/ch08.html)
- [x] Upgrade: Impact of this change on upgrade flows was considered and addressed if required
- [x] Testing: New code requires [new unit tests](https://github.com/kubevirt/kubevirt/blob/main/docs/reviewer-guide.md#when-is-a-pr-good-enough). New features and bug fixes require at least on e2e test
- [x] Documentation: A [user-guide update](https://github.com/kubevirt/user-guide/) was considered and is present (link) or not required. You want a user-guide update if it's a user facing feature / API change.
- [x] Community: Announcement to [kubevirt-dev](https://groups.google.com/g/kubevirt-dev/) was considered

### Release note
<!--  Write your release note:
1. Enter your extended release note in the below block. If the PR requires additional action from users switching to the new release, include the string "action required".
2. If no release note is required, just write "NONE".
-->
```release-note
Enable virt-exportproxy and virt-exportserver image for s390x
```

jschintag commented:
Testing confirms with this PR the x86 image for export(server|proxy) now correctly contains the x86 binary.
I will also check if the export of vms on s390x works, but i think that is secondary to putting a band aid on #12843

xpivarc reviewed (commented):
/approve

brianmcarey reviewed (commented):
This will fix it for exportproxy and exportserver but what about the other images that were not enabled for s390x - we still see double amd64 images available for these such as `network-passt-binding`

https://quay.io/repository/kubevirt/network-passt-binding/manifest/sha256:22d3524d38244f99b8f05b8382c7cd2bc66d8867748cafb41f38726a540d939f

jschintag commented:
> This will fix it for exportproxy and exportserver but what about the other images that were not enabled for s390x - we still see double amd64 images available for these such as `network-passt-binding`
> 
> https://quay.io/repository/kubevirt/network-passt-binding/manifest/sha256:22d3524d38244f99b8f05b8382c7cd2bc66d8867748cafb41f38726a540d939f

Yes, i'm currently looking into adopting the bazel-build-images, bazel-push-images and push-container-manifest scripts to only build enabled images per arch.

jschintag commented:
/hold

While i will still work to enable it, we will want to merge #12867 first to fix #12843 properly.
I will rebase+test this PR afterwards.

jschintag commented:
/unhold

I have rebased this PR on top of https://github.com/kubevirt/kubevirt/pull/12867 and verified that it works properly on s390x.

While rebasing i noticed that neither virt-export-server nor virt-export-proxy are build when running `make bazel-build-images`. Is that intended behavior or should i add them to it in a separate commit? For now i left it at the status quo.

mhenriks commented:
> /unhold
> 
> I have rebased this PR on top of #12867 and verified that it works properly on s390x.
> 
> While rebasing i noticed that neither virt-export-server nor virt-export-proxy are build when running `make bazel-build-images`. Is that intended behavior or should i add them to it in a separate commit? For now i left it at the status quo.

Pretty sure that is not intended I think it would make sense to add them

jschintag commented:
> > /unhold
> > I have rebased this PR on top of #12867 and verified that it works properly on s390x.
> > While rebasing i noticed that neither virt-export-server nor virt-export-proxy are build when running `make bazel-build-images`. Is that intended behavior or should i add them to it in a separate commit? For now i left it at the status quo.
> 
> Pretty sure that is not intended I think it would make sense to add them

I have added them in a second commit. I'm running the command locally right now to sanity check, since i'm not sure if it is run in the CI.

jschintag commented:
`make bazel-build-images` ran locally for me without trouble. Though it is hard to confirm if it build the image properly since it does not really output any artifacts i can find.

jschintag commented:
/retest-required

mhenriks commented on BUILD.bazel:
```diff
@@ -195,6 +195,8 @@ load("@bazel_skylib//lib:dicts.bzl", "dicts")
     # cmd images
     "$(container_prefix)/$(image_prefix)sidecar-shim:$(container_tag)": "//cmd/sidecars:sidecar-shim-image",
+    "$(container_prefix)/$(image_prefix)virt-exportproxy:$(container_tag)": "//cmd/virt-exportproxy:virt-exportproxy-image",
+    "$(container_prefix)/$(image_prefix)virt-exportserver:$(container_tag)": "//cmd/virt-exportserver:virt-exportserver-image",
```
Not sure that I would classify these as "other" images as they are created/managed by virt-operator. Was thinking this is a better place: https://github.com/jschintag/kubevirt/blob/virt-exportproxy/hack/bazel-build-images.sh#L53-L54

jschintag replied:
Yes, i was unsure where to add them. In the end i choose here, since that keeps the bash cli command looking cleaner.
But if you want them there @mhenriks i can move them to the script.

xpivarc replied:
Please move to `bazel-build-images.sh` where `virt-launcher` and such are build

jschintag replied:
I have moved the 2 to the script and verified locally that they are built.
Please check if it is as expected @xpivarc .

xpivarc reviewed (commented):
/approve
/hold
I have left one comment

jschintag commented:
/retest-required

mhenriks commented:
/lgtm

jschintag commented:
Since you added the hold @xpivarc is there anything still required from your site for the PR?

jschintag commented:
/retest-required

jschintag commented:
> > Since you added the hold @xpivarc is there anything still required from your site for the PR?
> 
> Sorry for taking so much time...

No worries, this PR is hardly a priority and i just came back from a short vacation anyway.
